### PR TITLE
Adds client.certFingerprint support plus unit tests.

### PR DESCRIPTION
**Pull request by rcrichton**
Closes https://github.com/jembi/openhim-core-js/issues/419

@Martin could you please review this for me?

### CODE REVIEW

File: test/spec/controllers/clientsmodal.js [hunk 1]
```diff
@@ -62,15 +72,33 @@ describe('Controller: ClientsmodalCtrl', function () {
     scope.client.name = '';
     scope.client.clientDomain = '';
     scope.client.roles = [];
-    scope.temp.password = 'password';
+    scope.temp.password = '';
 
     // run the validate
     scope.validateFormClients();
+    console.log(scope.ngError);
```
Comment (BMartinos) [inline test/spec/controllers/clientsmodal.js]: Remove console.log

File: test/spec/controllers/clientsmodal.js [hunk 2]
```diff
@@ -62,15 +72,33 @@ describe('Controller: ClientsmodalCtrl', function () {
...
+
+    scope.client.clientID = '';
+    scope.client.name = '';
+    scope.client.clientDomain = '';
+    scope.client.roles = [];
+    scope.temp.password = 'password';
+
+    // run the validate
+    scope.validateFormClients();
+    console.log(scope.ngError);
```
Comment (BMartinos) [inline test/spec/controllers/clientsmodal.js]: Remove console.log :)

**Comment by BMartinos** [comment]
Hey @rcrichton just two minor console.logs to be removed. Overall this looks great. One thing i noticed is that by default a certificate is selected. This also happens when editing a client that hasn't selected a certificate yet

File: app/views/clientsmodal.html [hunk 3]
```diff
@@ -57,6 +57,14 @@ <h4 class="modal-title" id="addClientLabel">
         </div>
       </div>
 
+      <div class="form-group" ng-class="{ 'has-error' : ngError.certFingerprint }">
+        <label><span>**</span> Client certificate</label>
```
Comment (BMartinos) [inline app/views/clientsmodal.html]: Maybe certificate with a capital C?  - "Certificate"

**Comment by rcrichton** [comment]
Cool, I've removed the console.logs :)

Are you running this with the new branch for core with the fingerprint changes? That branch should update with the correct values to make this work properly. If you are then it may be an issue.

File: app/views/clientsmodal.html (hunk 3 above)
Comment (rcrichton) [inline app/views/clientsmodal.html]: I'd rather leave it as it I think, I don't think it should be capitalised, its just a noun.

**Comment by BMartinos** [comment]
Sorry the issue with the certificate selection was on my side. I needed to use the core branch that upgraded my clients schema. This resolved the issue i noticed. All looks and functions great. :+1:

**Comment by BMartinos** [comment]
Good job on the upgrade btw :)

**Pull request merged**